components/motifs-view.js

```diff
@@ -0,0 +1,252 @@
+import PropTypes from "prop-types";
+import { useEffect, useState, useRef } from "react";
+import { useRouter } from "next/router";
+import { PlusCircleIcon, MinusCircleIcon } from "@heroicons/react/20/solid";
+import {
+  DataAreaTitle,
+  DataItemLabel,
+  DataItemValue,
+  DataPanel,
+} from "./data-area";
+import DnaLogo from "./dna-logo/dna-logo";
+
+/**
+ * Display information for a sigle motif. Each motif contains info about targets, strand, pwms and footprints, and a svg for DNA logo
+ */
+export function MotifElement({
+  motif,
+  windowStartPos,
+  windowEndPos,
+  relativeSnpCoordinate,
+}) {
+  const pwm = motif.dataMatrix;
+
+  // Determine padding required for alignment of logos
+  const motifStart = +motif.start;
+  const motifEnd = +motif.end;
+  const strand = motif.strand;
+  const startAlignmentNeeded =
+    strand === "-" ? windowEndPos - motifEnd : motifStart - +windowStartPos;
+  const endAlignmentNeeded =
+    strand === "-" ? motifStart - +windowStartPos : windowEndPos - motifEnd;
+  // Insert padding for alignment
+  const newPWM = [...pwm];
+  for (let startIdx = 0; startIdx < startAlignmentNeeded; startIdx += 1) {
+    newPWM.unshift([0, 0, 0, 0]);
+  }
+  for (let endIdx = 0; endIdx < endAlignmentNeeded; endIdx += 1) {
+    newPWM.push([0, 0, 0, 0]);
+  }
+  const targetList = motif.targets;
+  const pwmList = {};
+  const footprintList = {};
+
+  motif.datasets.forEach((datasetUrl, index) => {
+    const biosample = motif.biosamples[index];
+    const fileAccession = motif.accessions[index];
+    const datasetAccession = `accession=${
+      datasetUrl.split("/").slice(-2, -1)[0]
+    }`;
+
+    // So for a dataset, we get the Label and ID corresponding to the same index we are at now
+    if (biosample !== "") {
+      // If we have a biosample, then we map biosample to datasetAccession into footprintAnnotationList
+      if (footprintList[biosample]) {
+        footprintList[biosample].push(datasetAccession);
+      } else {
+        footprintList[biosample] = [datasetAccession];
+      }
+    } else {
+      // If we don't have a biosample, then map Label -> URL into `pwmList`
+      pwmList[fileAccession] = datasetUrl;
+    }
+  });
+
+  const footprintKeysSorted = Object.keys(footprintList).sort((a, b) =>
+    a.toLowerCase().localeCompare(b.toLowerCase())
+  );
+  const pwmsLength = Object.keys(pwmList).length;
+  const footprintsLength = Object.keys(footprintList).length;
+
+  const targetListLabel = targetList.indexOf(",") !== -1 ? "Targets" : "Target";
+  const pwmsLabel = pwmsLength > 1 ? "PWMs" : "PWM";
+  const footprintsLabel =
+    footprintsLength > 1 ? `Footprints (${footprintsLength})` : "Footprint";
```

Comment:
If `pwmsLabel` or `footprintsLabel` can contain 0, you have to make sure this produces the output you want.